core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1895,50 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager {
+    @volatile private var stopped = false
+    private val cacheReplicationThread = new Thread {
+      override def run(): Unit = {
+        while (blockManagerDecommissioning && !stopped) {
+          try {
+            logDebug(s"Attempting to replicate all cached RDD blocks")
+            offloadRddCacheBlocks()
+            logInfo(s"Attempt to replicate all cached blocks done")
+            val sleepInterval = if (Utils.isTesting) 100 else 30000
```

Comment:
To avoid magic number, shall we have a conf or a constant variable for this, at least, `30000`?

Reply:
done. Added a config which is overridden in tests.